modules/dnn/misc/python/test/test_dnn.py

```diff
@@ -53,6 +53,35 @@ def normAssertDetections(test, ref, out, confThreshold=0.0, scores_diff=1e-5, bo
     if errMsg:
         test.fail(errMsg)
 
+def normAssertDetections(test, refClassIds, refScores, refBoxes,
+                         testClassIds, testScores, testBoxes,
+                         confThreshold=0.0, scores_diff=1e-5, boxes_iou_diff=1e-4):
+    matchedRefBoxes = [False] * len(refBoxes)
+    errMsg = ''
+    for i in range(len(testBoxes)):
+        testScore = testScores[i]
+        if testScore < confThreshold:
+            continue
+
+        testClassId, testBox = testClassIds[i], testBoxes[i]
+        matched = False
+        for j in range(len(refBoxes)):
+            if (not matchedRefBoxes[j]) and testClassId == refClassIds[j] and \
+               abs(testScore - refScores[j]) < scores_diff:
+                interArea = inter_area(testBox, refBoxes[j])
+                iou = interArea / (area(testBox) + area(refBoxes[j]) - interArea)
+                if abs(iou - 1.0) < boxes_iou_diff:
+                    matched = True
+                    matchedRefBoxes[j] = True
+        if not matched:
+            errMsg += '\nUnmatched prediction: class %d score %f box %s' % (testClassId, testScore, box2str(testBox))
+
+    for i in range(len(refBoxes)):
+        if (not matchedRefBoxes[i]) and refScores[i] > confThreshold:
+            errMsg += '\nUnmatched reference: class %d score %f box %s' % (refClassIds[i], refScores[i], box2str(refBoxes[i]))
+    if errMsg:
+        test.fail(errMsg)
```

Comment:
Please avoid code duplication. Choose common parts with existing `normAssertDetections`. BTW there is no functions overloading in Python that means by adding second `def normAssertDetections` you replaces the first one.